src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/Microsoft.DotNet.UpgradeAssistant.Abstractions.csproj

```diff
@@ -17,6 +17,7 @@
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
     <PackageTags>Upgrade Assistant</PackageTags>
     <PackageIcon>icon.png</PackageIcon>
+    <DevelopmentDependency>true</DevelopmentDependency>
```

Comment:
What's this for? I'm not familiar with this property.

Reply:
This is so the resulting nupkg will automatically be added with `PrivateAssets=all` so they won't be in the output

Reply:
Wouldn't devs depending on the abstractions package want these assets in their output? These are typically UA extension scenarios and the common libraries will need to be present at run time, not just at build time like analzyers.

Is this change because UA will be loading the abstractions library from its own binaries in the default load context rather than from the extensions?

Reply:
Correct. It's a dev dependency precisely because we don't want it in the output. We need a unified type that we supply and don't want to accidently load another abstractions library in a different context. That was happening and causes weird type load exception.

We do guard against this explicitly by first checking if any type is already loaded in the main context, but since we don't need it there, it's just extra baggage.

Reply:
Makes sense. I have some anxiety about scenarios where someone implements something on top of the abstractions library and then uses it outside of UA (in which case the abstractions binaries maybe should be present) but I can't think of a likely scenario where that would happen, so this is probably good.

Reply:
Sure - they can always remove the private assets then if they really want to build on it themselves. This just helps the main use case to be in a good place.